routes/golfer_search.go

```diff
@@ -0,0 +1,64 @@
+package routes
+
+import (
+	"net/http"
+
+	"github.com/code-golf/code-golf/config"
+	"github.com/code-golf/code-golf/session"
+)
+
+type Solution struct {
```

Comment:
This type can move inside the func, it doesn't need to be visible at the package level. In fact with the sqlx Select() change it can be anonymous like `data`, no need to initialise the slice.